modules/fbx/data/fbx_mesh_data.cpp

```diff
@@ -262,51 +270,54 @@ MeshInstance *FBXMeshData::create_fbx_mesh(const ImportState &state, const FBXDo
 	{
 		PolygonId polygon_index = -1;
 		SurfaceId surface_id = -1;
-		SurfaceData *surface_data = nullptr;
 
 		for (size_t polygon_vertex = 0; polygon_vertex < polygon_indices.size(); polygon_vertex += 1) {
 			if (is_start_of_polygon(polygon_indices, polygon_vertex)) {
 				polygon_index += 1;
 				ERR_FAIL_COND_V_MSG(polygon_surfaces.has(polygon_index) == false, nullptr, "The FBX file is currupted, This surface_index is not expected.");
 				surface_id = polygon_surfaces[polygon_index];
-				surface_data = surfaces.getptr(surface_id);
-				CRASH_COND(surface_data == nullptr); // Can't be null.
 			}
 
+			// no copy and no ptr required.
+			SurfaceData &surface_data = surfaces[surface_id];
 			const int vertex = get_vertex_from_polygon_vertex(polygon_indices, polygon_vertex);
 
 			// The vertex position in the surface
 			// Uses a lookup table for speed with large scenes
-			bool has_polygon_vertex_index = surface_data->lookup_table.has(vertex);
+			bool has_polygon_vertex_index = surface_data.lookup_table.has(vertex);
 			int surface_polygon_vertex_index = -1;
 
 			if (has_polygon_vertex_index) {
-				surface_polygon_vertex_index = surface_data->lookup_table[vertex];
+				surface_polygon_vertex_index = surface_data.lookup_table[vertex];
 			} else {
-				surface_polygon_vertex_index = surface_data->vertices_map.size();
-				surface_data->lookup_table[vertex] = surface_polygon_vertex_index;
-				surface_data->vertices_map.push_back(vertex);
+				surface_polygon_vertex_index = surface_data.vertices_map.size();
+				surface_data.lookup_table[vertex] = surface_polygon_vertex_index;
+				surface_data.vertices_map.push_back(vertex);
 			}
 
-			surface_data->surface_polygon_vertex[polygon_index].push_back(surface_polygon_vertex_index);
+			surface_data.surface_polygon_vertex[polygon_index].push_back(surface_polygon_vertex_index);
 		}
 	}
 
 	//print_verbose("[debug UV 1] UV1: " + itos(uvs_0.size()));
 	//print_verbose("[debug UV 2] UV2: " + itos(uvs_1.size()));
 
 	// Phase 4. Per each surface just insert the vertices and add the indices.
-	for (const SurfaceId *surface_id = surfaces.next(nullptr); surface_id != nullptr; surface_id = surfaces.next(surface_id)) {
-		SurfaceData *surface = surfaces.getptr(*surface_id);
+	for (OrderedHashMap<SurfaceId, SurfaceData>::Element current_surface = surfaces.front(); current_surface; current_surface = current_surface.next()) {
```

Comment:
This can become local vector. Every index starts at zero for iterations.

Map, HashMap etc are overkill.

Reply:
Not really a good idea to edit this.